zebrad/src/components/mempool/crawler/tests.rs

```diff
@@ -70,4 +84,241 @@ proptest! {
             Ok::<(), TestCaseError>(())
         })?;
     }
+
+    /// Test if crawled transactions are forwarded to the [`Mempool`][mempool::Mempool] service.
+    ///
+    /// The transaction IDs sent by other peers to the crawler should be forwarded to the
+    /// [`Mempool`][mempool::Mempool] service so that they can be downloaded, verified and added to
+    /// the mempool.
+    #[test]
+    fn crawled_transactions_are_forwarded_to_downloader(
+        transaction_ids in vec(any::<UnminedTxId>(), 1..MAX_CRAWLED_TX),
+    ) {
+        let runtime = tokio::runtime::Builder::new_current_thread()
+            .enable_all()
+            .build()
+            .expect("Failed to create Tokio runtime");
+        let _guard = runtime.enter();
+
+        let transaction_id_count = transaction_ids.len();
+
+        runtime.block_on(async move {
+            let (mut peer_set, mut mempool, _sync_status, mut recent_sync_lengths) =
+                setup_crawler();
+
+            time::pause();
+
+            // Mock end of chain sync to enable the mempool crawler.
+            recent_sync_lengths.push_extend_tips_length(0);
```

Comment:
There is now a `mempool.enable()` test-only method that can be used to enable the mempool, but basically does the same thing

Reply:
Unfortunately I can't here, because the mempool service is actually a `MockService` :(

https://github.com/ZcashFoundation/zebra/blob/9cb74991debad7da734e759363c1128f3ea754ec/zebrad/src/components/mempool/crawler/tests.rs#L223.

Reply:
Oops :sweat_smile: 

I think you can use `SyncStatus::sync_close_to_tip` then (which is also test-only, and it's what `enable()` calls)

Reply:
Done in [0f6aebf](https://github.com/jvff/zebra/commit/0f6aebf540df20039cac08b2879cbaa6ca17d7a5), thanks!